In [78]:
import pandas as pd

In [79]:
#Set up variables for filepaths
cryptofilepath = "crypto_data.csv"

In [80]:
#pull file into dataframe

crypto_df = pd.read_csv(cryptofilepath, encoding="UTF-8")

In [81]:
#Review first 5 records of file
crypto_df.head()                 #1252 Records
#crypto_df.count()               #1252 Records, only 744 Mined - so delta is 508. 

Unnamed: 0   CoinName Algorithm  IsTrading ProofType  TotalCoinsMined  \
0         42    42 Coin    Scrypt       True   PoW/PoS     4.199995e+01   
1        365    365Coin       X11       True   PoW/PoS              NaN   
2        404    404Coin    Scrypt       True   PoW/PoS     1.055185e+09   
3        611  SixEleven   SHA-256       True       PoW              NaN   
4        808        808   SHA-256       True   PoW/PoS     0.000000e+00   

  TotalCoinSupply  
0              42  
1      2300000000  
2       532000000  
3          611000  
4               0

In [82]:
#Find records of that have isTrading =True
TradingCrypto_df=crypto_df.loc[crypto_df['IsTrading'] == True]

TradingCrypto_df.head()        #1144 Records
#TradingCrypto_df.count()        #685 Records

Unnamed: 0   CoinName Algorithm  IsTrading ProofType  TotalCoinsMined  \
0         42    42 Coin    Scrypt       True   PoW/PoS     4.199995e+01   
1        365    365Coin       X11       True   PoW/PoS              NaN   
2        404    404Coin    Scrypt       True   PoW/PoS     1.055185e+09   
3        611  SixEleven   SHA-256       True       PoW              NaN   
4        808        808   SHA-256       True   PoW/PoS     0.000000e+00   

  TotalCoinSupply  
0              42  
1      2300000000  
2       532000000  
3          611000  
4               0

In [83]:
#Remove IsTrading Column
del TradingCrypto_df['IsTrading']

In [84]:
# Checkscript
TradingCrypto_df.head()
#TradingCrypto_df.count()       #1144, 685 

Unnamed: 0   CoinName Algorithm ProofType  TotalCoinsMined TotalCoinSupply
0         42    42 Coin    Scrypt   PoW/PoS     4.199995e+01              42
1        365    365Coin       X11   PoW/PoS              NaN      2300000000
2        404    404Coin    Scrypt   PoW/PoS     1.055185e+09       532000000
3        611  SixEleven   SHA-256       PoW              NaN          611000
4        808        808   SHA-256   PoW/PoS     0.000000e+00               0

In [85]:
TradingCrypto_df = TradingCrypto_df.dropna(how='any',axis=0)

In [86]:
# Checkscript
TradingCrypto_df.head()
#TradingCrypto_df.count()      #685 Records across all 

Unnamed: 0   CoinName Algorithm ProofType  TotalCoinsMined TotalCoinSupply
0         42    42 Coin    Scrypt   PoW/PoS     4.199995e+01              42
2        404    404Coin    Scrypt   PoW/PoS     1.055185e+09       532000000
4        808        808   SHA-256   PoW/PoS     0.000000e+00               0
5       1337  EliteCoin       X13   PoW/PoS     2.927942e+10    314159265359
7        BTC    Bitcoin   SHA-256       PoW     1.792718e+07        21000000

In [87]:
#Identify records that are not unmined
TradingMinedCrypto_df = TradingCrypto_df[~TradingCrypto_df['TotalCoinsMined'].isnull()]
TradingMinedCrypto_df.head()        #685 Total Records
#TradingMinedCrypto_df.count()      #685 Total Records

Unnamed: 0   CoinName Algorithm ProofType  TotalCoinsMined TotalCoinSupply
0         42    42 Coin    Scrypt   PoW/PoS     4.199995e+01              42
2        404    404Coin    Scrypt   PoW/PoS     1.055185e+09       532000000
4        808        808   SHA-256   PoW/PoS     0.000000e+00               0
5       1337  EliteCoin       X13   PoW/PoS     2.927942e+10    314159265359
7        BTC    Bitcoin   SHA-256       PoW     1.792718e+07        21000000

In [88]:
#TradingMinedCrypto_df.head()

#Re-define datatype needed
TradingMinedCrypto_df["TotalCoinSupply"] = pd.to_numeric(TradingMinedCrypto_df["TotalCoinSupply"])
#TradingMinedCrypto_df.dtypes

#Find records of that have isTrading =True
TradingMinedNonZeroCrypto_df=TradingMinedCrypto_df.loc[(TradingMinedCrypto_df['TotalCoinSupply'] > 0)]

TradingMinedNonZeroCrypto_df.head()        #637 Records
#TradingMinedNonZeroCrypto_df.count()        #637 Records

Unnamed: 0   CoinName Algorithm ProofType  TotalCoinsMined  TotalCoinSupply
0         42    42 Coin    Scrypt   PoW/PoS     4.199995e+01     4.200000e+01
2        404    404Coin    Scrypt   PoW/PoS     1.055185e+09     5.320000e+08
5       1337  EliteCoin       X13   PoW/PoS     2.927942e+10     3.141593e+11
7        BTC    Bitcoin   SHA-256       PoW     1.792718e+07     2.100000e+07
9        LTC   Litecoin    Scrypt       PoW     6.303924e+07     8.400000e+07

In [89]:
#Checkscript
TradingMinedNonZeroCrypto_df.head() 

Unnamed: 0   CoinName Algorithm ProofType  TotalCoinsMined  TotalCoinSupply
0         42    42 Coin    Scrypt   PoW/PoS     4.199995e+01     4.200000e+01
2        404    404Coin    Scrypt   PoW/PoS     1.055185e+09     5.320000e+08
5       1337  EliteCoin       X13   PoW/PoS     2.927942e+10     3.141593e+11
7        BTC    Bitcoin   SHA-256       PoW     1.792718e+07     2.100000e+07
9        LTC   Litecoin    Scrypt       PoW     6.303924e+07     8.400000e+07

In [90]:
# Convert categorical data to numeric by 
# One-hot encoding the entire dataframe
CryptoOneHot_df = pd.get_dummies(TradingMinedNonZeroCrypto_df)
print(CryptoOneHot_df.columns)
CryptoOneHot_df

Index(['TotalCoinsMined', 'TotalCoinSupply', 'Unnamed: 0_1337',
       'Unnamed: 0_1CR', 'Unnamed: 0_404', 'Unnamed: 0_42', 'Unnamed: 0_AAC',
       'Unnamed: 0_ABJ', 'Unnamed: 0_ABS', 'Unnamed: 0_ABY',
       ...
       'ProofType_PoW/PoS', 'ProofType_PoW/PoS ', 'ProofType_PoW/PoW',
       'ProofType_PoW/nPoS', 'ProofType_Pos', 'ProofType_Proof of Authority',
       'ProofType_Proof of Trust', 'ProofType_TPoS',
       'ProofType_Zero-Knowledge Proof', 'ProofType_dPoW/PoW'],
      dtype='object', length=1378)


TotalCoinsMined  TotalCoinSupply  Unnamed: 0_1337  Unnamed: 0_1CR  \
0        4.199995e+01     4.200000e+01                0               0   
2        1.055185e+09     5.320000e+08                0               0   
5        2.927942e+10     3.141593e+11                1               0   
7        1.792718e+07     2.100000e+07                0               0   
9        6.303924e+07     8.400000e+07                0               0   
...               ...              ...              ...             ...   
1238     2.000000e+09     2.000000e+09                0               0   
1242     1.493105e+07     2.500000e+08                0               0   
1245     9.802226e+08     1.400223e+09                0               0   
1246     7.296538e+06     2.100000e+07                0               0   
1247     1.283270e+05     1.000000e+06                0               0   

      Unnamed: 0_404  Unnamed: 0_42  Unnamed: 0_AAC  Unnamed: 0_ABJ  \
0                  0              1               0               0   
2                  1              0               0               0   
5                  0              0               0               0   
7                  0              0               0               0   
9                  0              0               0               0   
...              ...            ...             ...             ...   
1238               0              0               0               0   
1242               0              0               0               0   
1245               0              0               0               0   
1246               0              0               0               0   
1247               0              0               0               0   

      Unnamed: 0_ABS  Unnamed: 0_ABY  ...  ProofType_PoW/PoS  \
0                  0               0  ...                  1   
2                  0               0  ...                  1   
5                  0               0  ...                  1   
7                  0               0  ...                  0   
9                  0               0  ...                  0   
...              ...             ...  ...                ...   
1238               0               0  ...                  0   
1242               0               0  ...                  1   
1245               0               0  ...                  0   
1246               0               0  ...                  0   
1247               0               0  ...                  0   

      ProofType_PoW/PoS   ProofType_PoW/PoW  ProofType_PoW/nPoS  \
0                      0                  0                   0   
2                      0                  0                   0   
5                      0                  0                   0   
7                      0                  0                   0   
9                      0                  0                   0   
...                  ...                ...                 ...   
1238                   0                  0                   0   
1242                   0                  0                   0   
1245                   0                  0                   0   
1246                   0                  0                   0   
1247                   0                  0                   0   

      ProofType_Pos  ProofType_Proof of Authority  ProofType_Proof of Trust  \
0                 0                             0                         0   
2                 0                             0                         0   
5                 0                             0                         0   
7                 0                             0                         0   
9                 0                             0                         0   
...             ...                           ...                       ...   
1238              0                             0                         0   
1242              0                           

This step set a multitude of binary columns in place of the named columns. We went from 6 columns to 1378 columns. 

In [91]:
#Remove CoinName Column from the dataframe
#del TradingMinedNonZeroCrypto_df['CoinName']
TradingMinedNonZeroCrypto_df.head()

Unnamed: 0   CoinName Algorithm ProofType  TotalCoinsMined  TotalCoinSupply
0         42    42 Coin    Scrypt   PoW/PoS     4.199995e+01     4.200000e+01
2        404    404Coin    Scrypt   PoW/PoS     1.055185e+09     5.320000e+08
5       1337  EliteCoin       X13   PoW/PoS     2.927942e+10     3.141593e+11
7        BTC    Bitcoin   SHA-256       PoW     1.792718e+07     2.100000e+07
9        LTC   Litecoin    Scrypt       PoW     6.303924e+07     8.400000e+07

In [97]:
# Scale the data in the fields that have large values
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
scaled_data = scaler.fit_transform(CryptoOneHot_df)

In [98]:
scaled_data

array([[-0.09782131, -0.03965512, -0.03965258, ..., -0.03965258,
        -0.03965258, -0.03965258],
       [-0.07228807, -0.03965497, -0.03965258, ..., -0.03965258,
        -0.03965258, -0.03965258],
       [ 0.61067897, -0.03956909, 25.21904043, ..., -0.03965258,
        -0.03965258, -0.03965258],
       ...,
       [-0.074102  , -0.03965474, -0.03965258, ..., -0.03965258,
        -0.03965258, -0.03965258],
       [-0.09764475, -0.03965511, -0.03965258, ..., -0.03965258,
        -0.03965258, -0.03965258],
       [-0.09781821, -0.03965512, -0.03965258, ..., -0.03965258,
        -0.03965258, -0.03965258]])

In [100]:
# Create a DataFrame with the transformed data
#FinalCrypto_df = pd.DataFrame(scaled_data)

# Create a DataFrame with the transformed data
FinalCrypto_df = pd.DataFrame(scaled_data, columns=CryptoOneHot_df.columns[:])

FinalCrypto_df.head()
#FinalCrypto_df.head()

TotalCoinsMined  TotalCoinSupply  Unnamed: 0_1337  Unnamed: 0_1CR  \
0        -0.097821        -0.039655        -0.039653       -0.039653   
1        -0.072288        -0.039655        -0.039653       -0.039653   
2         0.610679        -0.039569        25.219040       -0.039653   
3        -0.097388        -0.039655        -0.039653       -0.039653   
4        -0.096296        -0.039655        -0.039653       -0.039653   

   Unnamed: 0_404  Unnamed: 0_42  Unnamed: 0_AAC  Unnamed: 0_ABJ  \
0       -0.039653      25.219040       -0.039653       -0.039653   
1       25.219040      -0.039653       -0.039653       -0.039653   
2       -0.039653      -0.039653       -0.039653       -0.039653   
3       -0.039653      -0.039653       -0.039653       -0.039653   
4       -0.039653      -0.039653       -0.039653       -0.039653   

   Unnamed: 0_ABS  Unnamed: 0_ABY  ...  ProofType_PoW/PoS  ProofType_PoW/PoS   \
0       -0.039653       -0.039653  ...           1.420900           -0.039653   
1       -0.039653       -0.039653  ...           1.420900           -0.039653   
2       -0.039653       -0.039653  ...           1.420900           -0.039653   
3       -0.039653       -0.039653  ...          -0.703779           -0.039653   
4       -0.039653       -0.039653  ...          -0.703779           -0.039653   

   ProofType_PoW/PoW  ProofType_PoW/nPoS  ProofType_Pos  \
0          -0.039653           -0.039653      -0.039653   
1          -0.039653           -0.039653      -0.039653   
2          -0.039653           -0.039653      -0.039653   
3          -0.039653           -0.039653      -0.039653   
4          -0.039653           -0.039653      -0.039653   

   ProofType_Proof of Authority  ProofType_Proof of Trust  ProofType_TPoS  \
0                     -0.039653                 -0.039653       -0.039653   
1                     -0.039653                 -0.039653       -0.039653   
2                     -0.039653                 -0.039653       -0.039653   
3                     -0.039653                 -0.039653       -0.039653   
4                     -0.039653                 -0.039653       -0.039653   

   ProofType_Zero-Knowledge Proof  ProofType_dPoW/PoW  
0                       -0.039653           -0.039653  
1                       -0.039653           -0.039653  
2                       -0.039653           -0.039653  
3                       -0.039653           -0.039653  
4                       -0.039653           -0.039653  

[5 rows x 1378 columns]